### PR TITLE
[UT] add ut for utils/template.go

### PR DESCRIPTION
**Pull request by my-git9**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Add ut for utils/template.go
**Which issue(s) this PR fixes**:
Ref https://github.com/karmada-io/karmada/issues/2579

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/karmadactl/addons/utils/template_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	for _, tt := range tests {
+		got, err := ParseTemplate(tt.args.strTmpl, tt.args.obj)
+		if (err != nil) != tt.wantErr {
+			t.Errorf("ParseTemplate() error = %v, wantErr %v", err, tt.wantErr)
+			return
+		}
+
+		if (got == nil) != tt.wantNil {
+			t.Errorf("ParseTemplate() = %v , wantNil %v", got, tt.wantNil)
+		}
```
Comment (XiShanYongYe-Chang) [inline pkg/karmadactl/addons/utils/template_test.go:56]: Is it possible to return the content for an equivalent judgment?

Comment (my-git9) [reply pkg/karmadactl/addons/utils/template_test.go:56]: I will try it later, thx

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks
/lgtm~

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**